python/cudf/cudf/core/dataframe.py

```diff
@@ -2969,25 +3058,61 @@ def drop(
             outdf = self
         else:
             outdf = self.copy()
-        for c in columns:
-            outdf._drop_column(c)
-        return outdf
 
-    def drop_column(self, name):
-        """Drop a column by *name*
-        """
-        warnings.warn(
-            "The drop_column method is deprecated. "
-            "Use the drop method instead.",
-            DeprecationWarning,
-        )
-        self._drop_column(name)
+        def _drop_columns(df, columns, errors):
+            for c in columns:
+                try:
+                    df._drop_column(c)
+                except KeyError as e:
+                    if errors == "ignore":
+                        pass
+                    else:
+                        raise e
+
+        if axis in (1, "columns"):
+            _drop_columns(outdf, target, errors)
+        elif axis in (0, "index"):
+            index = cudf.Series(target)
```

Comment:
Does this have to be a Series? Do we need index handling of this object?

Reply:
This can be a column